### PR TITLE
Changes to drive autosigning

### PR DESCRIPTION
**Pull request by michaelcfanning**
I was able to create a successfully auto-signed build from these changes. Let's get them into master. We can merge everything back into a single powershell operation in the future. We will need to update our VSTS build tasks when that occurs (I can help with this).

@lgolding @Evmaus-MS

### CODE REVIEW

File: src/build.props [hunk 1]
```diff
@@ -35,9 +35,11 @@
   <PropertyGroup Label="Package">
     <PackageId Condition=" '$(PackageId)' == '' ">$(AssemblyName)</PackageId>
     <Authors Condition=" '$(Authors)' == '' ">$(Company)</Authors>
+    <Owners Condition=" '$(Authors)' == '' ">$(Company),tse-securitytools</Owners>
     <PackageRequireLicenseAcceptance Condition=" '$(PackageRequireLicenseAcceptance)' == '' ">false</PackageRequireLicenseAcceptance>
     <PackageLicenseUrl>https://github.com/Microsoft/jschema/blob/master/License.txt</PackageLicenseUrl>
     <PackageProjectUrl>https://github.com/Microsoft/jschema</PackageProjectUrl>
+    <PackageIconUrl>https://go.microsoft.com/fwlink/?linkid=2009431</PackageIconUrl>
```
Comment (lgolding) [inline src/build.props]: >microsoft [](start = 31, length = 9)

Any reason not to use the raw.github link?

File: CreateSigningDirectory.cmd [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+if "%Configuration%" EQU "" (
+set Configuration=Release
+)
+
+if "%Platform%" EQU "" (
+set Platform=AnyCPU
+)
+
+set BinaryOutputDirectory=%BinaryOutputDirectory%\%Platform%_%Configuration%\
+set SigningDirectory=%BinaryOutputDirectory%\..\Signing
```
Comment (lgolding) [inline CreateSigningDirectory.cmd]: >set [](start = 0, length = 3)

If you set `SigningDirectory` first, you won't have to back up a level :-)

Comment (lgolding) [reply CreateSigningDirectory.cmd]: The same in the other batch file.

---
In reply to: [209795461](https://github.com/Microsoft/jschema/pull/21#discussion_r209795461) [](ancestors = 209795461)

File: PushSignedBinariesIntoBuild.cmd [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+call :CopyFilesForMultitargeting Json.Schema.Validation.dll || goto :ExitFailed
+
+goto :Exit
+
+:CopyFilesForMultitargeting
+
+xcopy /Y %SigningDirectory%\netcoreapp2.0\Microsoft.%1 %BinaryOutputDirectory%\%~n1\netstandard2.0\
+if "%ERRORLEVEL%" NEQ "0" (echo %1 assembly copy failed. && goto :CopyFilesExit)
+
+echo xcopy /Y %SigningDirectory%\net461\Microsoft.%1 %BinaryOutputDirectory%\%~n1\net461\
```
Comment (lgolding) [inline PushSignedBinariesIntoBuild.cmd]: >echo [](start = 0, length = 4)

Remove `echo` (was presumably for debugging; doesn't appear below or in the other batch file).

**Comment by lgolding** [comment]
So the usage scenario is:

1.`BuildAndTest.cmd` (which creates unsigned packages containing unsigned binaries)
2. `CreateSigningDirectory.cmd`
3. Submit binaries for signing by some means not shown here.
4. `PushSignedBinariesIntoBuild.cmd`
5. Recreate packages from signed binaries by running `BuildAndTest.cmd -NoBuild -NoTest` (which just runs the packaging step).
6. Submit the packages themselves for signing by some means not shown here.

Presumably the missing steps are in the VSTS build definition.

Have I got that right?

File: PushSignedBinariesIntoBuild.cmd (hunk 3 above)
Comment (michaelcfanning) [reply PushSignedBinariesIntoBuild.cmd]: ok

---
In reply to: [209796642](https://github.com/Microsoft/jschema/pull/21#discussion_r209796642) [](ancestors = 209796642)

File: CreateSigningDirectory.cmd (hunk 2 above)
Comment (michaelcfanning) [reply CreateSigningDirectory.cmd]: right will fix this up.

---
In reply to: [209796248](https://github.com/Microsoft/jschema/pull/21#discussion_r209796248) [](ancestors = 209796248,209795461)

File: src/build.props (hunk 1 above)
Comment (michaelcfanning) [reply src/build.props]: hm was advised to use a forwarder, but probably not necessary to do so.

---
In reply to: [209795228](https://github.com/Microsoft/jschema/pull/21#discussion_r209795228) [](ancestors = 209795228)

Comment (lgolding) [reply src/build.props]: No, that's ok, I didn't know there was some standard practice in play here.

---
In reply to: [209798849](https://github.com/Microsoft/jschema/pull/21#discussion_r209798849) [](ancestors = 209798849,209795228)

**Comment by michaelcfanning** [comment]
Precisely. You only missed one detail, we pass -NoPackage on the first BuildAndTest call. Also, VSTS calls the powershell build script directly.

---
In reply to: [412709289](https://github.com/Microsoft/jschema/pull/21#issuecomment-412709289) [](ancestors = 412709289)

**Comment by lgolding** [comment]
Got it, all good. Ship it!

---
In reply to: [412711639](https://github.com/Microsoft/jschema/pull/21#issuecomment-412711639) [](ancestors = 412711639,412709289)

**Pull request merged**